### PR TITLE
Modify show tables pragma query to respect current catalog scope

### PR DESCRIPTION
**Pull request by rjatwal**
Addresses  https://github.com/duckdb/duckdb/issues/6812

### CODE REVIEW

File: src/function/pragma/pragma_queries.cpp [hunk 1]
```diff
@@ -15,7 +16,16 @@ string PragmaTableInfo(ClientContext &context, const FunctionParameters &paramet
...
 string PragmaShowTables(ClientContext &context, const FunctionParameters &parameters) {
-	return "SELECT name FROM sqlite_master ORDER BY name;";
+	auto catalog = DatabaseManager::GetDefaultDatabase(context);
+	auto table_query = "SELECT table_name as name FROM duckdb_tables where database_name='" + catalog + "'";
+	auto view_query = "SELECT view_name as name FROM duckdb_views where database_name='" + catalog + "'";
+	auto index_query = "SELECT index_name as name FROM duckdb_indexes where database_name='" + catalog + "'";
+
+	auto all_query = "select name from tables union all select name from views union all select name from indexes";
+
+	auto pragma_query = "with tables as ( " + table_query + "), views as ( " + view_query + +" ), indexes as ( " +
```
Comment (Tishj) [inline src/function/pragma/pragma_queries.cpp:26]: `view_query + +` that's a typo, no?
Could you please change the string creation in this PR to use StringUtil::Format?

You'll need to include `duckdb/common/string_util.hpp`

**Review by Tishj — CHANGES_REQUESTED** [review]

File: src/function/pragma/pragma_queries.cpp (hunk 1 above)
Comment (rjatwal) [reply src/function/pragma/pragma_queries.cpp:26]: 👍 Thanks for the comment that will indeed make code cleaner. Will update.

**Review by Mytherin — COMMENTED** [review]
Thanks for the PR! Could we perhaps also add a test that verifies the correct behavior of `SHOW TABLES`?

File: src/function/pragma/pragma_queries.cpp (hunk 1 above)
Comment (rjatwal) [reply src/function/pragma/pragma_queries.cpp]: Updated

**Comment by rjatwal** [comment]
Added test and extended to restricted to used schema as well

**Review by Mytherin — COMMENTED** [review]
Thanks for the fixes! Looks good to me. It just looks like there is an issue on Windows, likely because of a missing include:

```
2023-03-27T08:44:42.1694850Z D:\a\duckdb\duckdb\src\function\pragma\pragma_queries.cpp(22,23): error C2027: use of undefined type 'duckdb::ClientContext' [D:\a\duckdb\duckdb\src\function\pragma\duckdb_func_pragma.vcxproj]
```

Missing includes are often masked because of our unity builds (which concatenates all files in a single source directory as a single file). For windows (and our Expanded CI run) we do not do this. You can achieve the same by building with disable unity enabled - e.g. `DISABLE_UNITY=1 make`.

In this case you can also fix the issue by not looking into `context` directly, but by using `ClientData::Get(context)` instead.

**Comment by rjatwal** [comment]
@Mytherin 

For clang-tidy failure
```
/home/runner/work/duckdb/duckdb/src/function/pragma/pragma_queries.cpp:23:12: error: the 'empty' method should be used to check for emptiness instead of comparing to an empty object [readability-container-size-empty,-warnings-as-errors]
        schema = (schema == INVALID_SCHEMA) ? DEFAULT_SCHEMA : schema;
                  ^~~~~~~~~~~~~~~~~~~~~~~~
                  schema.empty()                 
``` 

Would you prefer I try to opt the line out of the clang-tidy rule or check for empty schema explicitly?

**Comment by Mytherin** [comment]
Ignoring is fine in this case - thanks!

**Pull request merged**